src/import/FSHImporter.ts

```diff
@@ -2059,6 +2069,67 @@ export class FSHImporter extends FSHVisitor {
     return path ? `${currentContext}.${path}` : currentContext;
   }
 
+  /**
+   * Given a code path and the context containing it, apply the code path context indicated by the code path's indent
+   * @param codePath - The code path to apply context to
+   * @param parentCtx - The parent element containing the code path
+   * @returns {string[]} - The code path with context prepended
+   */
+  private prependCodePathContext(codePath: string[], parentCtx: ParserRuleContext): string[] {
+    const location = this.extractStartStop(parentCtx);
+    const currentIndent = location.startColumn - this.baseIndent;
+
+    if (currentIndent > 0 && this.codePathContext.length === 0) {
+      logger.error(
+        'The first rule of a definition cannot be indented. The rule will be processed as if it is not indented.',
+        { location, file: this.currentFile }
+      );
+      return codePath;
+    }
```

Comment:
Maybe this block and the subsequent repeated blocks could be factored out into some sort of `validateContext` or `validateIndent` function? It just seems very similar to the three `if` statements` in the `prependPathContext` function. I think if you do that, it would be fine to drop the specific language of "code path" for simplicity.